test/test_functional.py

```diff
@@ -0,0 +1,202 @@
+import math
+
+import torch
+import torchaudio
+import unittest
+import test.common_utils
+
+
+class TestFunctional(unittest.TestCase):
+    data_sizes = [(2, 20), (3, 15)]
+    number_of_trials = 100
+    stored_rand_data = []
+    fixed_precision = int(1e10)
+
+    def setUp(self):
+        # we want to make sure that the random values are reproducible
+        self.stored_rand_data.clear()
+        torch.manual_seed(0)
+        for data_size in self.data_sizes:
+            rand_data1 = torch.randint(low=-self.fixed_precision, high=self.fixed_precision, size=data_size)
+            rand_data2 = torch.randint(low=-self.fixed_precision, high=self.fixed_precision, size=data_size)
+            self.stored_rand_data.append([rand_data1, rand_data2])
+
+    def _get_random_tensor(self, i):
+        # gets a random tensor of size data_sizes[i]. adds to previous tensors and then mods it.
+        rand_data = self.stored_rand_data[i]
```

Comment:
You don't need to store data for this. Take the first value (i), fix some constant to add (say c) and then some prime number (say p) to mod by.

So for element 0 you have i, for element 1 you have (i + c) mod  p for element 2 you have (element1 + c) mod p